lib/charms/nginx_ingress_integrator/v0/ingress.py

```diff
@@ -227,6 +231,11 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
         if not self.model.unit.is_leader():
             return
 
+        if not event.relation.data[event.app]:
+            LOGGER.info("Ingress hasn't finished configuring, waiting until relation is changed again.")
+            self.charm.on.ingress_proxy_available.emit()
```

Comment:
Are you sure you want to emit this event here? It is my understanding that, at this point, the ingress is not available.